plugins/repository-hdfs/src/test/java/org/elasticsearch/repositories/hdfs/HdfsBlobStoreContainerTests.java

```diff
@@ -18,16 +18,78 @@
  */
 package org.elasticsearch.repositories.hdfs;
 
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.AbstractFileSystem;
 import org.apache.hadoop.fs.FileContext;
+import org.apache.hadoop.fs.UnsupportedFileSystemException;
 import org.elasticsearch.common.blobstore.BlobStore;
 import org.elasticsearch.repositories.ESBlobStoreContainerTestCase;
 
+import javax.security.auth.Subject;
 import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.net.URI;
+import java.nio.file.Path;
+import java.security.AccessController;
+import java.security.Principal;
+import java.security.PrivilegedAction;
+import java.util.Collections;
 
 public class HdfsBlobStoreContainerTests extends ESBlobStoreContainerTestCase {
 
     @Override
     protected BlobStore newBlobStore() throws IOException {
-        return new HdfsBlobStore(FileContext.getFileContext(), "", 100);
+        Path tempDir = createTempDir(null);
+        FileContext fileContext = AccessController.doPrivileged(
+                new PrivilegedAction<FileContext>() {
+                    @Override
+                    public FileContext run() {
+                        return createContext(tempDir.toUri());
+            }
+        });
+
+        return new HdfsBlobStore(fileContext, "temp", 100);
+    }
+
+    public FileContext createContext(URI uri) {
+        Configuration cfg = new Configuration(true);
+        cfg.setClassLoader(HdfsRepository.class.getClassLoader());
+        cfg.reloadConfiguration();
```

Comment:
I'm afraid I can't be of much assistance right now. The HDFS integration tests are running against a miniHDFS cluster which is triggered as a fixture (thanks to Ryan and RobertM). Trying to mock Hadoop is likely to be unsuccessful since it's quite greedy in terms of permissions (last I checked, when pointing to a folder, it tried to read the whole tree from the root...).
I think there's still a `TestingFS` that tries to alleviate some of the problems but I'm not sure to what degree it is used.
Double check the permissions created at startup - you might need to create a folder _inside_ the temp folder.